### PR TITLE
Observation/plannedTime

### PR DESCRIPTION
**Pull request by tpolecat**
This adds the `plannedTime` field to `type Observation`.

### CODE REVIEW

File: modules/service/src/main/scala/lucuma/odb/graphql/mapping/NonNegDurationMapping.scala [hunk 1]
```diff
@@ -9,24 +9,28 @@ import edu.gemini.grackle.skunk.SkunkMapping
...
-        LargeProgramType / "totalTime"       -> nonNegDurationMapping(ProposalTable.TotalTime)(ProposalTable.ProgramId),
-        AllocationType / "duration"          -> nonNegDurationMapping(AllocationTable.Duration)(AllocationTable.ProgramId, AllocationTable.Partner),
+        ProgramType / "plannedTime" / "pi"            -> nonNegDurationMapping(ProgramTable.PlannedTime.Pi)(ProgramTable.Id),
+        ProgramType / "plannedTime" / "uncharged"     -> nonNegDurationMapping(ProgramTable.PlannedTime.Uncharged)(ProgramTable.Id),
+        ProgramType / "plannedTime" / "execution"     -> nonNegDurationMapping(ProgramTable.PlannedTime.Execution)(ProgramTable.Id),
+        ObservationType / "plannedTime" / "pi"        -> nonNegDurationMapping(ObservationView.PlannedTime.Pi)(ObservationView.Id),
+        ObservationType / "plannedTime" / "uncharged" -> nonNegDurationMapping(ObservationView.PlannedTime.Uncharged)(ObservationView.Id),
+        ObservationType / "plannedTime" / "execution" -> nonNegDurationMapping(ObservationView.PlannedTime.Execution)(ObservationView.Id),
```
Comment (tpolecat) [inline modules/service/src/main/scala/lucuma/odb/graphql/mapping/NonNegDurationMapping.scala:30]: This is where I needed the more general `SwitchMapping` from #204

File: modules/service/src/main/resources/db/migration/V0030__observations.sql [hunk 2]
```diff
@@ -150,6 +150,11 @@ create table t_observation (
   c_active_status      e_obs_active_status not null    default 'active',
   c_visualization_time timestamp           null        default null,
 
+  -- planned time summary
+  c_pts_pi        interval not null default '0h',
+  c_pts_uncharged interval not null default '0h',
+  c_pts_execution interval not null default '0h',
```
Comment (swalker2m) [inline modules/service/src/main/resources/db/migration/V0030__observations.sql:156]: This seems fine for now.  How might it eventually be implemented?  It'll have to be a fairly complicated calculation that we won't be able to directly execute in a stored procedure.  Many different kinds of updates to the observation would have to also update these values.

**Review by swalker2m — APPROVED** [review]
Looks good.  :+1:

File: modules/service/src/main/resources/db/migration/V0030__observations.sql (hunk 2 above)
Comment (tpolecat) [reply modules/service/src/main/resources/db/migration/V0030__observations.sql:156]: Unclear. We may need to recompute it when we do various updates, or just do it on-demand as a top-level computation.

**Pull request merged**